pkg/generators/system/config/options.go

```diff
@@ -1,191 +1,84 @@
+	opts.Unpack(spec.Config.SearchServer.BatchSize).IntoEnvvar("THINKING_SPHINX_BATCH_SIZE")
+
+	opts.Unpack(spec.Config.DatabaseDSN).IntoEnvvar("DATABASE_URL").AsSecretRef("system-database")
+
+	opts.Unpack(spec.Config.MemcachedServers).IntoEnvvar("MEMCACHE_SERVERS")
+
+	opts.Unpack(spec.Config.Recaptcha.PublicKey).IntoEnvvar("RECAPTCHA_PUBLIC_KEY").AsSecretRef("system-recaptcha")
+	opts.Unpack(spec.Config.Recaptcha.PrivateKey).IntoEnvvar("RECAPTCHA_PRIVATE_KEY").AsSecretRef("system-recaptcha")
+
+	opts.Unpack(spec.Config.EventsSharedSecret).IntoEnvvar("EVENTS_SHARED_SECRET").AsSecretRef("system-events-hook")
+
+	opts.Unpack(spec.Config.Redis.QueuesDSN).IntoEnvvar("REDIS_URL")
+	opts.Unpack("").IntoEnvvar("REDIS_NAMESPACE")
+	opts.Unpack("").IntoEnvvar("REDIS_SENTINEL_HOSTS")
+	opts.Unpack("").IntoEnvvar("REDIS_SENTINEL_ROLE")
+
+	opts.Unpack(spec.Config.SMTP.Address).IntoEnvvar("SMTP_ADDRESS")
+	opts.Unpack(spec.Config.SMTP.User).IntoEnvvar("SMTP_USER_NAME").AsSecretRef("system-smtp")
+	opts.Unpack(spec.Config.SMTP.Password).IntoEnvvar("SMTP_PASSWORD").AsSecretRef("system-smtp")
+	opts.Unpack(spec.Config.SMTP.Port).IntoEnvvar("SMTP_PORT")
+	opts.Unpack(spec.Config.SMTP.AuthProtocol).IntoEnvvar("SMTP_AUTHENTICATION")
+	opts.Unpack(spec.Config.SMTP.OpenSSLVerifyMode).IntoEnvvar("SMTP_OPENSSL_VERIFY_MODE")
+	opts.Unpack(spec.Config.SMTP.STARTTLS).IntoEnvvar("SMTP_STARTTLS")
+	opts.Unpack(spec.Config.SMTP.STARTTLSAuto).IntoEnvvar("SMTP_STARTTLS_AUTO")
+
+	opts.Unpack(spec.Config.MappingServiceAccessToken).IntoEnvvar("APICAST_ACCESS_TOKEN").AsSecretRef("system-master-apicast")
+
+	opts.Unpack(spec.Config.Zync.Endpoint).IntoEnvvar("ZYNC_ENDPOINT")
+	opts.Unpack(spec.Config.Zync.AuthToken).IntoEnvvar("ZYNC_AUTHENTICATION_TOKEN").AsSecretRef("system-zync")
+
+	opts.Unpack(spec.Config.Backend.RedisDSN).IntoEnvvar("BACKEND_REDIS_URL")
+	opts.Unpack("").IntoEnvvar("BACKEND_REDIS_SENTINEL_HOSTS")
+	opts.Unpack("").IntoEnvvar("BACKEND_REDIS_SENTINEL_ROLE")
+	opts.Unpack(spec.Config.Backend.InternalEndpoint).IntoEnvvar("BACKEND_ROUTE") // DEPRECATED
+	opts.Unpack(spec.Config.Backend.InternalEndpoint).IntoEnvvar("BACKEND_URL")
+	opts.Unpack(spec.Config.Backend.ExternalEndpoint).IntoEnvvar("BACKEND_PUBLIC_URL")
+	opts.Unpack(spec.Config.Backend.InternalAPIUser).IntoEnvvar("CONFIG_INTERNAL_API_USER").AsSecretRef("system-backend")
+	opts.Unpack(spec.Config.Backend.InternalAPIPassword).IntoEnvvar("CONFIG_INTERNAL_API_PASSWORD").AsSecretRef("system-backend")
+
+	opts.Unpack(spec.Config.Assets.AccessKey).IntoEnvvar("AWS_ACCESS_KEY_ID").AsSecretRef("system-multitenant-assets-s3")
+	opts.Unpack(spec.Config.Assets.SecretKey).IntoEnvvar("AWS_SECRET_ACCESS_KEY").AsSecretRef("system-multitenant-assets-s3")
+	opts.Unpack(spec.Config.Assets.Bucket).IntoEnvvar("AWS_BUCKET")
+	opts.Unpack(spec.Config.Assets.Region).IntoEnvvar("AWS_REGION")
+	opts.Unpack(spec.Config.Assets.Host).IntoEnvvar("RAILS_ASSET_HOST")
```

Comment:
we had some logic for assets.Hosts, where if not set in the Cr, we set explicitly `""`, I guess the effect of current implementation does not change? (it was for dev only I guess)

Reply:
It doesn't affect anything as I also added a cloudfront dist for the dev-eng environment some weeks ago. Right now the envvar won't be set if the value is not set in the CR.